### PR TITLE
[New Rule] Dumping Accounts Hashes using Built-In Commands

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/907

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/credential_access_dumping_hashes_bi_cmds.toml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+creation_date = "2021/01/25"
+maturity = "production"
+updated_date = "2021/01/25"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the execution of macOS built-in commands to dump user account hashes. Adversaries may attempt to dump
+credentials to obtain account login and credential material in the form of a hash and crack them or leverage techniques
+such as pass the hash for lateral movement.
```
Comment (peasead) [inline rules/macos/credential_access_dumping_hashes_bi_cmds.toml:11]:
```suggestion
Identifies the execution of macOS built-in commands used to dump user account hashes. Adversaries may attempt to dump
credentials to obtain account login and credential material in the form of a hash. The hashes can be cracked or leveraged for lateral movement.
```

File: rules/macos/credential_access_dumping_hashes_bi_cmds.toml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+]
+risk_score = 73
+rule_id = "02ea4563-ec10-4974-b7de-12e65aa4f9b3"
+severity = "high"
+tags = ["Elastic", "Host", "macOS", "Threat Detection", "Credential Access"]
+type = "query"
+
+query = '''
+event.category:process and event.type:(start or process_started) and
+ process.name:(defaults or mkpassdb) and process.args:(ShadowHashData or "-dump")
```
Comment (peasead) [inline rules/macos/credential_access_dumping_hashes_bi_cmds.toml:30]:
```suggestion
event.category:process and event.type:start and
 process.name:(defaults or mkpassdb) and process.args:(ShadowHashData or "-dump")
```

File: rules/macos/credential_access_dumping_hashes_bi_cmds.toml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+[metadata]
+creation_date = "2021/01/25"
+maturity = "production"
+updated_date = "2021/01/25"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the execution of macOS built-in commands used to dump user account hashes. Adversaries may attempt to dump
+credentials to obtain account login and credential material in the form of a hash. These hashes can be cracked or leveraged for lateral movement.
```
Comment (bm11100) [inline rules/macos/credential_access_dumping_hashes_bi_cmds.toml:10]:
```suggestion
Identifies the execution of macOS built-in commands used to dump user account hashes. Adversaries may attempt to dump
credentials to obtain account login information in the form of a hash. These hashes can be cracked or leveraged for lateral movement.
```

File: rules/macos/credential_access_dumping_hashes_bi_cmds.toml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Dumping Accounts Hashes using Built-In Commands"
```
Comment (bm11100) [inline rules/macos/credential_access_dumping_hashes_bi_cmds.toml:16]:
```suggestion
name = "Dumping Account Hashes via Built-In Commands"
```

File: rules/macos/credential_access_dumping_hashes_bi_cmds.toml (hunk 3 above)
Comment (bm11100) [reply rules/macos/credential_access_dumping_hashes_bi_cmds.toml:10]: re-lint

**Review by bm11100 — APPROVED** [review]

**Review by threat-punter — APPROVED** [review]

**Pull request merged**